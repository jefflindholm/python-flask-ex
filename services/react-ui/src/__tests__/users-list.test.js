import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import UsersList from '../components/users-list';

const users = [
  { active: true, email: 'jeff@lindholm.org', id: 1, username: 'jeff' },
  { active: true, email: 'iryna@lindholm.org', id: 2, username: 'iryna' },
];

test('UsersList renders properly', () => {
  const wrapper = shallow(<UsersList users={users}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.className).toBe('card card-body bg-light');
  expect(element.get(0).props.children).toBe('jeff');
});

test('UsersList renders correct snapshot', () => {
  const tree = renderer.create(<UsersList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
