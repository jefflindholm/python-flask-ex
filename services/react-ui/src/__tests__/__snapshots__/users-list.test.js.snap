// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders correct snapshot 1`] = `
<div>
  <h4
    className="card card-body bg-light"
  >
    jeff
  </h4>
  <h4
    className="card card-body bg-light"
  >
    iryna
  </h4>
</div>
`;
